File: ECommerce.Core/Entities/Category.cs
using System;
using System.Collections.Generic;

namespace ECommerce.Core.Entities;

public class Category : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string Slug { get; set; } = string.Empty;
    public string? ImageUrl { get; set; }
    
    public string? MetaTitle { get; set; }
    public string? MetaDescription { get; set; }
    
    public int DisplayOrder { get; set; }
    public bool IsActive { get; set; } = true;

    public int? ParentId { get; set; }
    public Category? ParentCategory { get; set; }
    public ICollection<Category> ChildCategories { get; set; } = new List<Category>();

    public ICollection<SubCategory> SubCategories { get; set; } = new List<SubCategory>();
    public ICollection<Product> Products { get; set; } = new List<Product>();
}

File: ECommerce.View/src/app/core/models/category.ts
export interface Category {
  id: string;
  name: string;
  slug: string;
  href?: string;
  parentId?: string | null;
  imageUrl?: string;
  isActive: boolean;
  productCount?: number;
  displayOrder?: number;
  metaTitle?: string;
  metaDescription?: string;
  subCategories?: Category[];
}

export interface SubCategory {
  id: string;
  name: string;
  slug: string;
  categoryId: string;
  isActive: boolean;
  imageUrl?: string;
  displayOrder?: number;
  collections?: Collection[];
}

export interface Collection {
  id: string;
  name: string;
  slug: string;
  subCategoryId: string;
  isActive: boolean;
}

export interface CategoryNode {
  category: Category;
  children: CategoryNode[];
}

export interface ReorderPayload {
  parentId: number | null;
  orderedIds: number[];
}